### PR TITLE
Unit test least used balancer

### PR DESCRIPTION
**Pull request by JamesNK**
Tests to verify the experimental least used load balancer policy.

### CODE REVIEW

File: test/FunctionalTests/Balancer/BalancerHelpers.cs [hunk 1]
```diff
@@ -164,6 +167,53 @@ public static async Task WaitForChannelStateAsync(ILogger logger, GrpcChannel ch
             logger.LogInformation($"{channelId}: Current channel state '{currentState}' matches expected state '{state}'.");
         }
 
+        public static async Task<Subchannel[]> WaitForSubChannelsToBeReadyAsync(ILogger logger, GrpcChannel channel, int expectedCount, Func<SubchannelPicker?, Subchannel[]>? getPickerSubchannels = null)
+        {
+            if (getPickerSubchannels == null)
+            {
+                getPickerSubchannels = (picker) =>
+                {
+                    if (picker is RoundRobinPicker roundRobinPicker)
```
Comment (captainsafia) [inline test/FunctionalTests/Balancer/BalancerHelpers.cs:176]: #PranavPoints?

```csharp
return picker switch
{
  RoundRobinPicker roundRobinPicker => roundRobinPicker._subchannels.ToArray(),
  PickFirstPicker pickFirstPicker => new[] { pickFirstPicker.Subchannel },
  null => Array.Empty<Subchannel>(),
  _ => throw new Exception("Unexpected picker type: " + picker.GetType().FullName)
};
```

File: test/FunctionalTests/Balancer/LeastUsedBalancerTests.cs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+#if SUPPORT_LOAD_BALANCING
+#if NET5_0_OR_GREATER
```
Comment (captainsafia) [inline test/FunctionalTests/Balancer/LeastUsedBalancerTests.cs:20]: Do we need this check here if `SUPPORT_LOAD_BALANCING` is only set when the `TargetFramework` is greater than `net5.0`?

File: test/FunctionalTests/Balancer/BalancerHelpers.cs [hunk 3]
```diff
@@ -122,7 +123,9 @@ public static async Task<GrpcChannel> CreateChannel(ILoggerFactory loggerFactory
             var services = new ServiceCollection();
             services.AddSingleton<ResolverFactory>(new TestResolverFactory(resolver));
             services.AddSingleton<IRandomGenerator>(new TestRandomGenerator());
+            services.AddSingleton<IRandomGenerator>(new TestRandomGenerator());
```
Comment (captainsafia) [inline test/FunctionalTests/Balancer/BalancerHelpers.cs:126]: Why do we inject this twice here? It seems like we always resolve a single instance anyways?

Comment (JamesNK) [reply test/FunctionalTests/Balancer/BalancerHelpers.cs]: Copy/paste typo

**Review by captainsafia — APPROVED** [review]
LGTM conflicts aside.

**Pull request merged**